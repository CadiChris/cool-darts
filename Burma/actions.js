import Contrats from "./Contrats";

export const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
export const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  payload: {
    nomDuJoueur
  }
});

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = () => ({
  type: DEMARRER_PARTIE
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (
  lanceur,
  nombreDeSimplesBull,
  nombreDeDoublesBull
) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeSimplesBull, nombreDeDoublesBull)
  }
});
