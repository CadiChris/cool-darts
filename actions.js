const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  payload: {
    nomDuJoueur
  }
});

const DEMARRER_PARTIE = "DEMARRER_PARTIE";
const demarrerPartie = () => ({
  type: DEMARRER_PARTIE
});

const VOLEE_SUR_CHIFFRE = "VOLEE_SUR_CHIFFRE";
const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE_SUR_CHIFFRE,
  payload: {
    lanceur,
    chiffre,
    nombreDeTouches
  }
});

const VOLEE_SUR_DOUBLE = "VOLEE_SUR_DOUBLE";
const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE_SUR_DOUBLE,
  payload: {
    lanceur,
    chiffresTouches
  }
});

const VOLEE_SUR_TRIPLE = "VOLEE_SUR_TRIPLE";
const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE_SUR_TRIPLE,
  payload: {
    lanceur,
    chiffresTouches
  }
});

const VOLEE_SUR_BULL = "VOLEE_SUR_BULL";
const voleeSurBull = (lanceur, nombreDeSimplesBull, nombreDeDoublesBull) => ({
  type: VOLEE_SUR_BULL,
  payload: {
    lanceur,
    nombreDeSimplesBull,
    nombreDeDoublesBull
  }
});

export {
  INSCRIRE_JOUEUR,
  inscrireJoueur,
  DEMARRER_PARTIE,
  demarrerPartie,
  VOLEE_SUR_CHIFFRE,
  VOLEE_SUR_DOUBLE,
  VOLEE_SUR_TRIPLE,
  VOLEE_SUR_BULL,
  voleeSurChiffre,
  voleeSurDouble,
  voleeSurTriple,
  voleeSurBull
};
