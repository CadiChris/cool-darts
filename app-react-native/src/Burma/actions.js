import Contrats from "./arbitrage/Contrats";

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = joueurs => ({
  type: DEMARRER_PARTIE,
  joueurs
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (lanceur, nombreDeBulls) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeBulls)
  }
});
