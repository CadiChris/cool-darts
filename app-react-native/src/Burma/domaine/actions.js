import Contrats from './arbitrage/Contrats';

export const DEMARRER_BURMA = 'BURMA/DEMARRER';
export const demarrerBurma = joueurs => ({
  type: DEMARRER_BURMA,
  joueurs,
});

export const VOLEE = 'BURMA/VOLEE';
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches),
  },
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches),
  },
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches),
  },
});
