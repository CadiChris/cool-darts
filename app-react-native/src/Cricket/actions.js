export const demarrerPartie = joueurs => ({
  type: "DEMARRER_PARTIE",
  joueurs
});

export const lancerFlechette = (joueur, chiffre, touches) => ({
  type: "LANCER_FLECHETTE",
  joueur,
  chiffre,
  touches
});

export const nouvellePartie = () => ({
  type: "NOUVELLE_PARTIE"
});
