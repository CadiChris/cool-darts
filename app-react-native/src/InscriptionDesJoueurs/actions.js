export const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
export const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  nomDuJoueur
});

export const DESINSCRIRE_JOUEUR = "DESINSCRIRE_JOUEUR";
export const desinscrireJoueur = nomDuJouer => ({
  type: DESINSCRIRE_JOUEUR,
  nomDuJouer
});

export const CHOISIR_JEU = "CHOISIR_JEU";
export const choisirJeu = jeuChoisi => ({
  type: CHOISIR_JEU,
  jeuChoisi
});
