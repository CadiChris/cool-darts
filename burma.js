import { INSCRIRE_JOUEUR, DEMARRER_PARTIE, VOLEE } from "./burma.actions";
import * as lanceur from "./lanceur";
import { scores } from "./scores";
import { leChiffreSuivant } from "./chiffre";

const burma = (state = STATE_INITIAL, action) => {
  switch (action.type) {
    case INSCRIRE_JOUEUR:
      return {
        ...state,
        joueurs: [...state.joueurs, action.payload.nomDuJoueur],
        scores: scores(state.scores, action)
      };

    case DEMARRER_PARTIE:
      return {
        ...state,
        lanceur: state.joueurs[0],
        scores: scores(state.scores, action),
        phase: "EN_COURS",
        chiffreCourant: 15
      };

    case VOLEE:
      return {
        ...state,
        lanceur: lanceur.suivant(state.joueurs, action.payload.lanceur),
        chiffreCourant: leChiffreSuivant(state.chiffreCourant).avec(
          state.joueurs,
          action.payload.lanceur
        ),
        scores: scores(state.scores, action),
        phase: derniereVolee(
          state.joueurs,
          action.payload.lanceur,
          action.payload.chiffre
        )
          ? "TERMINEE"
          : state.phase
      };

    default:
      return state;
  }
};

const derniereVolee = (tousLesJoueurs, lanceur, chiffre) =>
  tousLesJoueurs.indexOf(lanceur) === tousLesJoueurs.length - 1 &&
  chiffre === "B";

const STATE_INITIAL = {
  joueurs: [],
  scores: scores(undefined, {}),
  phase: "INSCRIPTION",
  lanceur: undefined,
  chiffreCourant: undefined
};

export { burma };
