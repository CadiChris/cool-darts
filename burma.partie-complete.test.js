import { burma } from "./burma";
import { inscrireJoueur, demarrerPartie, volee } from "./burma.actions";

it("déroule une partie complète", () => {
  const burma1joueur = burma(undefined, inscrireJoueur("J1"));
  const burma2joueurs = burma(burma1joueur, inscrireJoueur("J2"));
  const burmaEnCours = burma(burma2joueurs, demarrerPartie());

  const v_15_j1 = burma(burmaEnCours, volee("J1", 15, 0));
  expect(v_15_j1).toMatchSnapshot();

  const v_15_j2 = burma(v_15_j1, volee("J2", 15, 2));
  expect(v_15_j2).toMatchSnapshot();
});
