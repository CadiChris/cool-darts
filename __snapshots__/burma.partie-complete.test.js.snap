// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`déroule une partie complète 1`] = `
Object {
  "chiffreCourant": 15,
  "joueurs": Array [
    "J1",
    "J2",
  ],
  "lanceur": "J2",
  "phase": "EN_COURS",
  "scores": Array [
    Object {
      "joueur": "J1",
      "points": 20,
      "touches": Object {
        "15": 0,
      },
    },
    Object {
      "joueur": "J2",
      "points": 40,
      "touches": Object {},
    },
  ],
  "vainqueur": undefined,
}
`;

exports[`déroule une partie complète 2`] = `
Object {
  "chiffreCourant": 16,
  "joueurs": Array [
    "J1",
    "J2",
  ],
  "lanceur": "J1",
  "phase": "EN_COURS",
  "scores": Array [
    Object {
      "joueur": "J1",
      "points": 20,
      "touches": Object {
        "15": 0,
      },
    },
    Object {
      "joueur": "J2",
      "points": 70,
      "touches": Object {
        "15": 2,
      },
    },
  ],
  "vainqueur": undefined,
}
`;
